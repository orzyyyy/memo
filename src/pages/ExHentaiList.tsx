import React from 'react';
import { ExHentaiInfoItem } from '../../server/controller/ExhentaiController';
import { Row, Col, Card, Dropdown, Menu } from 'antd';
import LazyLoad from 'react-lazyload';

export interface DownloadProps {
  url: string;
  name?: string;
}
export interface ExHentaiListProps {
  dataSource: ExHentaiInfoItem[] | null;
  onDownload: ({ url, name }: DownloadProps) => void;
}

const openDetail = (url: string) => {
  window.open(url);
};

export default ({ dataSource, onDownload }: ExHentaiListProps) => (
  <Row gutter={16} style={{ width: '100%' }}>
    {dataSource &&
      dataSource.map((item, i) => (
        <Col span={4} key={item.detailUrl + i}>
          <LazyLoad height={document.body.clientHeight} once>
            <Dropdown
              overlay={
                <Menu>
                  <Menu.Item
                    key="download"
                    onClick={() => {
                      onDownload({ url: item.detailUrl });
                    }}
                  >
                    download
                  </Menu.Item>
                </Menu>
              }
              trigger={['contextMenu']}
            >
              <Card
                hoverable
                style={{ height: document.body.clientHeight / 2 }}
                onClick={() => openDetail(item.detailUrl)}
              >
                <img alt={item.name} src={item.thumbnailUrl} />
              </Card>
            </Dropdown>
          </LazyLoad>
        </Col>
      ))}
  </Row>
);
