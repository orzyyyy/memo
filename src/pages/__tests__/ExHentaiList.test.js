import React from 'react';
import { shallow } from 'enzyme';
import ExHentaiList from '../ExHentaiList';

describe('ExHentaiList', () => {
  const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});

  afterAll(() => {
    errorSpy.mockRestore();
  });

  it('render correctly', () => {
    const wrapper = shallow(
      <ExHentaiList
        dataSource={[
          {
            name:
              '[Zakotsu] Amanita Pantherina (COMIC Koh 2018-06) [Chinese] [嗷呜个人汉化] [Digital]',
            detailUrl: 'https://exhentai.org/g/1430261/17651c12eb/',
            postTime: 1559412420000,
            thumbnailUrl:
              'https://exhentai.org/t/5b/47/5b47c48abf0fb6c685869520cdb8ab0d97990e4d-1074410-1280-1807-jpg_250.jpg',
          },
        ]}
        wrapperHeight={1000}
        onDownload={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
