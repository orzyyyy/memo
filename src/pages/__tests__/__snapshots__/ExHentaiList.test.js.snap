// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExHentaiList render correctly 1`] = `
<Row
  gutter={16}
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Col
    key="https://exhentai.org/g/1430261/17651c12eb/0"
    span={4}
  >
    <LazyLoad
      height={0}
      offset={0}
      once={true}
      overflow={false}
      resize={false}
      scroll={true}
      unmountIfInvisible={false}
    >
      <Dropdown
        mouseEnterDelay={0.15}
        mouseLeaveDelay={0.1}
        overlay={
          <Menu>
            <MenuItem
              onClick={[Function]}
            >
              download
            </MenuItem>
          </Menu>
        }
        placement="bottomLeft"
        trigger={
          Array [
            "contextMenu",
          ]
        }
      >
        <Card
          hoverable={true}
          onClick={[Function]}
          style={
            Object {
              "height": 0,
            }
          }
        >
          <img
            alt="[Zakotsu] Amanita Pantherina (COMIC Koh 2018-06) [Chinese] [嗷呜个人汉化] [Digital]"
            src="https://exhentai.org/t/5b/47/5b47c48abf0fb6c685869520cdb8ab0d97990e4d-1074410-1280-1807-jpg_250.jpg"
          />
        </Card>
      </Dropdown>
    </LazyLoad>
  </Col>
</Row>
`;
